packages/react/src/BackboneElementInput/BackboneElementInput.tsx

```diff
@@ -22,33 +23,35 @@ export interface BackboneElementInputProps {
 export function BackboneElementInput(props: BackboneElementInputProps): JSX.Element {
   const [defaultValue] = useState(() => props.defaultValue ?? {});
   const parentElementsContext = useContext(ElementsContext);
-  const profileUrl = props.profileUrl ?? parentElementsContext.profileUrl;
+  const profileUrl = props.profileUrl ?? parentElementsContext?.profileUrl;
   const typeSchema = useMemo(() => tryGetDataType(props.typeName, profileUrl), [props.typeName, profileUrl]);
   const type = typeSchema?.type ?? props.typeName;
 
-  const elementsContext = useMemo(() => {
+  const contextValue: ElementsContextType | undefined = useMemo(() => {
+    if (!typeSchema) {
+      return undefined;
+    }
     return buildElementsContext({
       parentContext: parentElementsContext,
-      elements: typeSchema?.elements,
-      parentPath: props.path,
-      parentType: type,
-      profileUrl,
+      elements: typeSchema.elements,
+      path: props.path,
+      profileUrl: typeSchema.url,
     });
-  }, [parentElementsContext, typeSchema?.elements, props.path, type, profileUrl]);
+  }, [typeSchema, props.path, parentElementsContext]);
 
   if (!typeSchema) {
     return <div>{type}&nbsp;not implemented</div>;
   }
 
-  return (
-    <ElementsContext.Provider value={elementsContext}>
-      <ElementsInput
-        path={props.path}
-        type={type}
-        defaultValue={defaultValue}
-        onChange={props.onChange}
-        outcome={props.outcome}
-      />
-    </ElementsContext.Provider>
+  return maybeWrapWithContext(
+    ElementsContext.Provider,
+    contextValue,
```

Comment:
Trying to parse when `contextValue === undefined` here

The result of the `useMemo` will only be undefined when `typeSchema` is undefined, yeah?

And we're already checking for `!typeSchema` above?

So we could probably add `if (!typeSchema || !contextValue)` to that check?

Reply:
Maybe the naming needs to be improved or a more explicit helper function needs to be added, but `buildElementsContext` can return `undefined` as well if it detects that the context it is being asked to build would be identical to it's parent context.